src/main/kotlin/org/rust/lang/core/types/infer/Declarations.kt

```diff
@@ -59,13 +59,7 @@ fun inferTypeReferenceType(type: RsTypeReference, defaultTraitObjectRegion: Regi
                         TyTraitObject(listOfNotNull(boundElement.downcast()), defaultTraitObjectRegion ?: ReUnknown)
                     }
                     target is RsTypeDeclarationElement -> {
-                        val ty = target.declaredType
-                            .substituteWithTraitObjectRegion(subst, defaultTraitObjectRegion ?: ReStatic)
-                        if (ty is TyAdt && ty.item != target && target is RsTypeAlias) {
```

Comment:
Why this approach doesn't work. At first glance, it looks like removing `ty is TyAdt` should be enough.

Reply:
Well, to be honest, I didn't even try it, it seemed like it would be *too easy*. It indeed seems to work :D